src/core_plugins/elasticsearch/lib/check_es_version.js

```diff
@@ -1,31 +1,70 @@
+/**
+ * ES and Kibana versions are locked, so Kibana should require that ES has the same version as
+ * that defined in Kibana's package.json.
+ */
+
 import _ from 'lodash';
 import esBool from './es_bool';
-import versionSatisfies from './version_satisfies';
+import parseVersionNumbers from './parse_version_numbers';
+import isEsCompatibleWithKibana from './is_es_compatible_with_kibana';
+import isEsNewerThanKibana from './is_es_newer_than_kibana';
 import SetupError from './setup_error';
 
-module.exports = function (server) {
+module.exports = function checkEsVersion(server, kibanaVersion) {
   server.log(['plugin', 'debug'], 'Checking Elasticsearch version');
 
   const client = server.plugins.elasticsearch.client;
-  const engineVersion = server.config().get('elasticsearch.engineVersion');
 
   return client.nodes.info()
   .then(function (info) {
-    const badNodes = _.filter(info.nodes, function (node) {
-      // remove nodes that satify required engine version
-      return !versionSatisfies(node.version, engineVersion);
-    });
+    // Aggregate incompatible ES nodes.
+    const incompatibleNodes = [];
+
+    // Aggregate ES nodes which should prompt a Kibana upgrade.
+    const warningNodes = [];
 
-    if (!badNodes.length) return true;
+    _.forEach(info.nodes, esNode => {
+      if (!isEsCompatibleWithKibana(esNode.version, kibanaVersion)) {
+        return incompatibleNodes.push(esNode);
+      }
 
-    const badNodeNames = badNodes.map(function (node) {
-      return 'Elasticsearch v' + node.version + ' @ ' + node.http_address + ' (' + node.ip + ')';
+      // It's acceptable if ES is ahead of Kibana, but we want to prompt users to upgrade Kibana
+      // to match it.
+      if (isEsNewerThanKibana(esNode.version, kibanaVersion)) {
+        warningNodes.push(esNode);
+      }
     });
 
-    const message = `This version of Kibana requires Elasticsearch ` +
-    `${engineVersion} on all nodes. I found ` +
-    `the following incompatible nodes in your cluster: ${badNodeNames.join(',')}`;
+    function getEsNodeNames(nodes) {
+      return nodes.map(node => {
+        return 'Elasticsearch v' + node.version + ' @ ' + node.http_address + ' (' + node.ip + ')';
+      });
+    }
+
+    if (warningNodes.length) {
+      const warningNodeNames = getEsNodeNames(warningNodes);
+
+      const warningMessage =
+        `This version of Kibana (${kibanaVersion}) is behind some of the versions of Elasticsearch ` +
+        `on the nodes in your cluster. Kibana is behind the following nodes in your cluster: ` +
+        `${warningNodeNames.join(',')}`;
+
+      server.log(['warning'], {
+        tmpl: warningMessage,
```

Comment:
`tmpl` lets you add additional data that isn't pretty, but would be helpful in analyzing the logs. The template can include `<%= %>` variable interpolation, and reads from the rest of the object passed as the log message. This data is ignored when the log is formatted as text, but is included when the message is formatted as json.

Consider this alternate approach:

```js
const mapNodeToSimpleNode = node => (
  { version: node.version, http_address: node.http.publish_address, ip: node.ip }
);

const mapSimpleNodeToName = node =>
  'v' + node.version + ' @ ' + node.http_address + ' (' + node.ip + ')';

if (warningNodes.length) {
  server.log(['warning'], {
    tmpl: (
      'This version of Kibana (<%= kibanaVersion %>) is behind some of the versions of Elasticsearch ' +
      'on the nodes in your cluster. Kibana is behind the following nodes in your cluster: ' +
      '<%= nodes.map(mapSimpleNodeToName).join(", ") %>'
    ),
    kibanaVersion,
    mapSimpleNodeToName,
    nodes: warningNodes.map(mapNodeToSimpleNode)
  });
}
```

In text this just looks like the formatted template:
![image](https://cloud.githubusercontent.com/assets/1329312/18463214/2d4157f0-793d-11e6-97cb-8e494cd4ac63.png)

But in JSON, it includes all the node data:
```json
{
  "type": "log",
  "@timestamp": "2016-09-13T06:00:31Z",
  "tags": [
    "warning"
  ],
  "pid": 46273,
  "kibanaVersion": "5.0.0-alpha1",
  "warningNodes": [
    {
      "version": "5.1.0",
      "http_address": "127.0.0.1:9200",
      "ip": "127.0.0.1"
    }
  ],
  "message": "This version of Kibana (5.0.0-alpha1) is behind some of the versions of Elasticsearch on the nodes in your cluster. Kibana is behind the following nodes in your cluster: v5.1.0 @ 127.0.0.1:9200 (127.0.0.1)"
}
```